pydantic_ai_slim/pydantic_ai/models/xai.py

```diff
@@ -0,0 +1,1046 @@
+
+        # Insert instructions as a system message at the beginning if present
+        if instructions:
+            xai_messages.insert(0, system(instructions))
+
+        # Convert tools: combine built-in (server-side) tools and custom (client-side) tools
+        tools: list[chat_types.chat_pb2.Tool] = []
+        if model_request_parameters.builtin_tools:
+            tools.extend(self._get_builtin_tools(model_request_parameters))
+        if model_request_parameters.tool_defs:
+            tools.extend(self._map_tools(model_request_parameters))
+        tools_param = tools if tools else None
+
+        # Set tool_choice based on whether tools are available and text output is allowed
+        profile = GrokModelProfile.from_profile(self.profile)
+        if not tools:
+            tool_choice: Literal['none', 'required', 'auto'] | None = None
+        elif not model_request_parameters.allow_text_output and profile.grok_supports_tool_choice_required:
+            tool_choice = 'required'
+        else:
+            tool_choice = 'auto'
+
+        # Set response_format based on the output_mode
+        response_format: chat_pb2.ResponseFormat | None = None
+        if model_request_parameters.output_mode == 'native':
+            output_object = model_request_parameters.output_object
+            assert output_object is not None
+            response_format = self._map_json_schema(output_object)
+        elif (
+            model_request_parameters.output_mode == 'prompted' and not tools and profile.supports_json_object_output
+        ):  # pragma: no branch
+            response_format = self._map_json_object()
+
+        # Map model settings to xAI SDK parameters
+        xai_settings = self._map_model_settings(model_settings)
+
+        # Populate use_encrypted_content and include based on model settings
+        include: list[chat_pb2.IncludeOption] = []
+        use_encrypted_content = False
+        if profile.grok_supports_encrypted_reasoning_content and model_settings.get('xai_include_encrypted_content'):
+            use_encrypted_content = True
+        if model_settings.get('xai_include_code_execution_outputs'):
+            include.append(chat_pb2.IncludeOption.INCLUDE_OPTION_CODE_EXECUTION_CALL_OUTPUT)
+        if model_settings.get('xai_include_web_search_outputs'):
+            include.append(chat_pb2.IncludeOption.INCLUDE_OPTION_WEB_SEARCH_CALL_OUTPUT)
+        if model_settings.get('xai_include_x_search_outputs'):
+            include.append(chat_pb2.IncludeOption.INCLUDE_OPTION_X_SEARCH_CALL_OUTPUT)
+        if model_settings.get('xai_include_document_search_outputs'):
+            include.append(chat_pb2.IncludeOption.INCLUDE_OPTION_DOCUMENT_SEARCH_CALL_OUTPUT)
+        if model_settings.get('xai_include_collections_search_outputs'):
+            include.append(chat_pb2.IncludeOption.INCLUDE_OPTION_COLLECTIONS_SEARCH_CALL_OUTPUT)
```

Comment:
See above; we should only include the options if we actually give users a way to access that information